sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/BaseAuthResolverHandler.cs

```diff
@@ -65,9 +67,66 @@ protected void PreInvoke(IExecutionContext executionContext)
             }
         }
 
+        /// <summary>
+        /// Certain allow-listed services contain auth schemes in their endpoint ruleset. For those, we'll attempt to
+        /// retrieve the auth options from the attributes of the resolved endpoint.
+        /// </summary>
+        protected static List<IAuthSchemeOption> RetrieveSchemesFromEndpoint(Endpoint endpoint)
+        {
+            if (endpoint == null || endpoint.Attributes == null)
+            {
+                return null;
+            }
+
+            var authSchemes = endpoint.Attributes["authSchemes"] as IList;
+            if (authSchemes == null)
+            {
+                return null;
+            }
+
+            var schemeNames = authSchemes.OfType<PropertyBag>().Select(scheme => (string)scheme["name"]).ToList();
+
+            // If there's only one scheme, we'll try to return one of the default collections to avoid extra allocations.
+            if (schemeNames.Count == 1)
+            {
+                var schemeName = schemeNames.First();
+                switch (schemeName)
+                {
+                    case "sigv4":
+                    case "sigv4-s3express":
+                        return AuthSchemeOption.DEFAULT_SIGV4;
+                    case "sigv4a":
+                        return AuthSchemeOption.DEFAULT_SIGV4A;
+                    default:
```

Comment:
Can we add the other auth scheme options preemptively even if they aren't used now? maybe atleast ad the noAuth version?